src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -726,13 +726,32 @@ static int get_pool_offset(sd_dhcp_server *server, be32_t requested_ip) {
         return be32toh(requested_ip & ~server->netmask) - server->pool_offset;
 }
 
+static int prepare_new_lease(DHCPLease **lease, be32_t address, DHCPClientId *client_id, uint8_t *chaddr, be32_t gateway, usec_t expiration){
+        *lease = new0(DHCPLease, 1);
+        if (!lease)
+                return -ENOMEM;
+        (*lease)->address = address;
+        (*lease)->client_id.data = memdup(client_id->data, client_id->length);
+        if (!(*lease)->client_id.data) {
+                free(*lease);
+                return -ENOMEM;
+        }
+        (*lease)->client_id.length = client_id->length;
+        memcpy(&(*lease)->chaddr, chaddr, ETH_ALEN);
+        (*lease)->gateway = gateway;
+        (*lease)->expiration = expiration;
```

Comment:
please use C99 structured initialization to initialize most fields here.

i.e.

```c
_cleanup_(dhcp_lease_unrefp) DHCPLease *lease = NULL;
lease = new(DHCPLease, 1);
if (!lease)
        return -ENOMEM;
*lease = (DHCPLease) {
        .gateway = gateway,
        .expiration = expiration,
        …
};
…
*ret_lease = TAKE_PTR(lease);
```